lib/devise/failure_app.rb

```diff
@@ -142,11 +142,16 @@ def scope_url
 
       opts[:format] = request_format unless skip_format?
 
-      opts[:script_name] = relative_url_root if relative_url_root?
-
       router_name = Devise.mappings[scope].router_name || Devise.available_router_name
       context = send(router_name)
 
+      if relative_url_root?
+        opts[:script_name] = relative_url_root
+      elsif defined? context.routes
+        rootpath = context.routes.url_helpers.root_path
+        opts[:script_name] = rootpath.chomp('/') unless rootpath.length <= 1
```

Comment:
My brain doesn't process `unless` very well 🙈. I always prefer an `if` with the positive case, for me, it's easier to read. So I'd suggest to change it to:
```ruby
opts[:script_name] = rootpath.chomp('/') if rootpath.length > 1
```

Wdyt?

Reply:
I do agree. `if` is to be preferred, so please go with it.